internal/services/instance/servers_data_source.go

```diff
@@ -165,8 +165,8 @@ func DataSourceInstanceServersRead(ctx context.Context, d *schema.ResourceData,
 	for _, server := range res.Servers {
 		rawServer := make(map[string]interface{})
 		rawServer["id"] = zonal.NewID(server.Zone, server.ID).String()
-		if server.PublicIP != nil {
-			rawServer["public_ip"] = server.PublicIP.Address.String()
+		if server.PublicIP != nil { //nolint:staticcheck
```

Comment:
We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.